include/ews/ews.hpp

```diff
@@ -8092,6 +8153,51 @@ static_assert(std::is_move_constructible<mailbox>::value, "");
 static_assert(std::is_move_assignable<mailbox>::value, "");
 #endif
 
+struct directory_id
+{
+    std::string id;
+};
+
+#ifdef EWS_HAS_NON_BUGGY_TYPE_TRAITS
+static_assert(std::is_default_constructible<directory_id>::value, "");
+static_assert(std::is_copy_constructible<directory_id>::value, "");
+static_assert(std::is_copy_assignable<directory_id>::value, "");
+static_assert(std::is_move_constructible<directory_id>::value, "");
+static_assert(std::is_move_assignable<directory_id>::value, "");
+#endif
+
+struct resolution
+{
+    ews::mailbox mailbox;
+    ews::directory_id directory_id;
+};
+
+#ifdef EWS_HAS_NON_BUGGY_TYPE_TRAITS
+static_assert(std::is_default_constructible<resolution>::value, "");
+static_assert(std::is_copy_constructible<resolution>::value, "");
+static_assert(std::is_copy_assignable<resolution>::value, "");
+static_assert(std::is_move_constructible<resolution>::value, "");
+static_assert(std::is_move_assignable<resolution>::value, "");
+#endif
+
+struct resolution_set
+{
+    int indexed_paging_offset = 0;
+    int numerator_offset = 0;
+    int absolute_denominator = 0;
+    bool includes_last_item_in_range = true;
+    int total_items_in_view = 0;
+    std::vector<resolution> resolutions;
+};
```

Comment:
I don't think that default initializers work with VS2012. You really want to add a c'tor set here.